client/landing/gutenboarding/index.tsx

```diff
@@ -144,3 +150,28 @@ function waitForSelectedSite(): Promise< Site | undefined > {
 		select( SITE_STORE ).getSite( selectedSite );
 	} ).finally( unsubscribe );
 }
+/**
+ * If available-designs-config.json has been updated, replace the cached list
+ * of designs with the updated designs
+ */
+function checkIfRandomizedDesignsAreUpToDate() {
+	const designsInStore = select( ONBOARD_STORE ).getRandomizedDesigns();
+	if ( ! isUpToDate( designsInStore.featured, availableDesigns.featured ) ) {
+		dispatch( ONBOARD_STORE ).setRandomizedDesigns( {
+			...availableDesigns,
+			featured: shuffle( availableDesigns.featured ),
+		} );
+	}
+}
+
+/**
+ *
+ * Compare cached designs in the ONBOARD_STORE to the source of designs in
+ * available-designs-config.json
+ *
+ * @param stored randomizedDesigns cached in WP_ONBOARD
+ * @param available designs sourced from available-designs-config.json
+ */
+function isUpToDate( stored: Design[], available: Design[] ): boolean {
```

Comment:
Maybe there is a better home for this? Like in a utility file?

Also this needs a unit test, on my next-to-do list for this PR.

Reply:
We do have a `lib` sub-directory we can use if it makes sense to move the function elsewhere. Because it's only used in one place right now, and it's directly related to `checkIfRandomizedDesignsAreUpToDate`, I quite like leaving it here right next to the function that calls it. It means there's not far to go to look up what the function does!